protobuf/lib/src/protobuf/coded_buffer.dart

```diff
@@ -33,8 +33,19 @@ void _mergeFromCodedBufferReader(BuilderInfo meta, _FieldSet fs,
     CodedBufferReader input, ExtensionRegistry registry) {
   ArgumentError.checkNotNull(registry);
   fs._ensureWritable();
+
+  // Micro-optimization: cache the storage lookup for repeated fields.
+  var prevTag = -1;
+  List? cachedList;
+
   while (true) {
     final tag = input.readTag();
+    // If the current field's tag is different from previous, invalidate cache.
+    if (tag != prevTag) {
+      cachedList = null;
+      prevTag = tag;
+    }
```

Comment:
In my benchmarking, caching the list actually makes it slower when compiled to JS with `dart2js -O4`.

It seems to make a small improvement in AOT so can still keep it, but I'm not sure if it's worth it.

See my top-level comment for details.